### PR TITLE
[aio] Fixed AIO read slow on ARC

### PR DESCRIPTION
**Pull request by jimmy-huang**
The adc_read() on Zephyr has a latency of 100ms which causes
the AIO read to be very slow.  This patch now spawns a thread
that fetch ADC for any pins that are enabled so that
the main thread on ARC will just look up the fetched value.

Signed-off-by: Jimmy Huang <jimmy.huang@intel.com>

### CODE REVIEW

**Comment by grgustaf** [comment]
Is this a bug in Zephyr?

**Comment by jimmy-huang** [comment]
@grgustaf I think so, because it's adc_read() that's taking so long, that's a native zephyr call

**Comment by jimmy-huang** [comment]
https://www.zephyrproject.org/doc/1.7.0/kernel/common/common_atomic.html

It does say that it support a 32bit value, no?

**Comment by jprestwo** [comment]
@jimmy-huang, ah yes your right. LGTM.

File: scripts/analyze.sh [hunk 1]
```diff
@@ -525,7 +525,10 @@ if check_for_feature "Accelerometer\|Gyroscope\|AmbientLightSensor\|TemperatureS
             fi
         fi
         if $sensor_light; then
+            # Ambient Light sensor only works with AIO at the moment
+            MODULES+=" -DBUILD_MODULE_AIO"
             echo "CONFIG_ADC=y" >> $ARCPRJFILE
+            echo "export ZJS_AIO=y" >> $CONFFILE
```
Comment (grgustaf) [inline scripts/analyze.sh]: You'll presumably have to fix this after the JSON patch goes in.

**Comment by jimmy-huang** [comment]
@grgustaf rebased

**Comment by grgustaf** [comment]
+1

**Pull request merged**